araproc/analysis/standard_reco.py

```diff
@@ -599,3 +599,61 @@ def calculate_surface_corr_ratio(self, *maps, z_thresh=-10):
         
         return surf_corr_ratio, max_surf_corr_result, max_corr_result
 
+    def min_frac_corr_depth(self, corr_map, fraction=0.6, z_thresh=-10):
+        """
+        Finds the shallowest depth where correlation meets a specified fraction of the maximum correlation,
+        using 'theta' to calculate depth and adjusting by the average z-coordinate of antennas.
+        Ensures the result remains below the specified depth threshold (z_thresh).
+
+        Parameters
+        ----------
+        corr_map : dict
+            Contains:
+            - 'map': ROOT.TH2D histogram with correlation values by theta and phi.
+            - 'radius': Correlation radius in meters.
+        fraction : float
+            Fraction of the max correlation to set the threshold.
+            Default is 0.6 (or 60%).
+        z_thresh : float
+            Maximum allowable depth (relative to the surface). Default is -10 meters.
+
+        Returns
+        -------
+        min_depth : float
+            Shallowest depth where correlation meets the specified fraction of max, adjusted by avg z, and below z_thresh.
```

Comment:
Maybe this can be more clear. Just "Shallowest depth (measured from ice surface) where correlation meets the specified fraction of max and below z_thresh."